packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -669,6 +670,40 @@ export class MirrorNodeClient {
         requestId);
     }
 
+    /**
+    * Check if transaction fail is because of contract revert and try to fetch and log the reason.
+    *
+    * @param e
+    * @param requestId
+    * @param requestIdPrefix
+    */
+    public async getContractRevertReasonFromTransaction(e: any, requestId: string | undefined, requestIdPrefix: string): Promise<any | undefined> {
+        if (e instanceof SDKClientError && e.isContractRevertExecuted()) {
+            const transactionId = e.message.match(constants.TRANSACTION_ID_REGEX);
+            if (transactionId) {
+              const tx = await this.getTransactionById(transactionId[0], undefined, requestId);
+
+              if(tx === null){
+
+                this.logger.error(`${requestIdPrefix} Transaction failed with null result: ${MirrorNodeClientError.statusCodes.NOT_FOUND}`);
+                return MirrorNodeClientError.statusCodes.NOT_FOUND;
```

Comment:
Return null here.
`NOT_FOUND` would be if we just made an HTTP request call but for this function we want the message so it it's invalid return null

Reply:
Done